synapse/storage/databases/main/user_directory.py

```diff
@@ -222,12 +212,13 @@ def _get_next_batch(
             if not rooms_to_work_on:
                 return None
 
-            # Get how many are left to process, so we can give status on how
-            # far we are in processing
-            txn.execute("SELECT COUNT(*) FROM " + TEMP_TABLE + "_rooms")
-            result = txn.fetchone()
-            assert result is not None
-            progress["remaining"] = result[0]
+            if "remaining" not in progress:
```

Comment:
I don't understand what this is trying to do. Does this mean we write a progress count at the end of the first iteration and never update the counter?

Reply:
Ah sorry. We fetch the count once, and then afterwards we decrement the count every time we delete a row from the table.

https://github.com/matrix-org/synapse/blob/f519534cc1c880f4a85bd88cba31b462f38b9578/synapse/storage/databases/main/user_directory.py#L314:L314

Reply:
Ah, perfect.